rust/src/lib.rs

```diff
@@ -779,7 +790,9 @@ pub unsafe extern "C" fn zcashlc_list_transparent_receivers(
         let db_data = unsafe { wallet_db(db_data, db_data_len, network)? };
         let account_uuid = account_uuid_from_bytes(account_uuid_bytes)?;
 
-        match db_data.get_transparent_receivers(account_uuid) {
+        // This API is documented as returning the transparent receivers for allocated
+        // diversified UAs, which means change taddrs are excluded.
+        match db_data.get_transparent_receivers(account_uuid, false) {
```

Comment:
I think the intention is to eventually pass both external and change addresses over the same FFI call, or maybe expose this boolean across it. I'm not doing that in this PR as it's focused on the Rust updates and MSRV bump; I want this to be separately testable against the Swift SDK before we start making the transparent-related and address-rotation-related FFI changes (plus this is a big PR already).

Reply:
This is the correct change for the current PR.